### PR TITLE
Improve shutdown of VertxHttpExporter and VertxGrpcExporter

### PR DESCRIPTION
**Pull request by jimkont**
improves shutdown of VertxHttpExporter and VertxGrpcExporter but setting the CompletableResultCode only when the corresponding http/grpc client close() is complete.

- Related to https://github.com/quarkusio/quarkus/pull/38895

### CODE REVIEW

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxHttpExporter.java [hunk 1]
```diff
@@ -173,8 +183,26 @@ public byte[] responseBody() {
 
         @Override
         public CompletableResultCode shutdown() {
-            client.close();
-            return CompletableResultCode.ofSuccess();
+            if (!isShutdown.compareAndSet(false, true)) {
+                logger.log(Level.INFO, "Calling shutdown() multiple times.");
```
Comment (brunobat) [inline extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxHttpExporter.java:187]: We probably should place the log level in debug.

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java [hunk 2]
```diff
@@ -138,10 +139,24 @@ public CompletableResultCode flush() {
     public CompletableResultCode shutdown() {
         if (!isShutdown.compareAndSet(false, true)) {
             logger.log(Level.INFO, "Calling shutdown() multiple times.");
```
Comment (brunobat) [inline extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/exporter/otlp/VertxGrpcExporter.java:141]: We probably should place the log level in debug.

**Review by geoand — APPROVED** [review]
Thanks!

**Review by brunobat — APPROVED** [review]
Thanks @jimkont

**Pull request merged**